library/types/src/modules/String.rb

```diff
@@ -369,14 +358,15 @@ def Repeat(text, number)
     # @param [Symbol] alignment alignment to use, either `left or `right
     # @return padded text
     def SuperPad(text, length, padding, alignment)
-      text = "" if text.nil?
+      text ||= ""
+      return text if length.nil? || text.size >= length || padding.nil?
 
-      pad = Repeat(padding, Ops.subtract(length, Builtins.size(text)))
+      pad = padding * (length - text.size)
```

Comment:
well, then it won't be backward compatible. But I agree that it looks like buggy ( old code also do repetition ).